pyfolio/tears.py

```diff
@@ -1536,9 +1536,15 @@ def create_perf_attrib_tear_sheet(returns,
     perf_attrib.show_perf_attrib_stats(returns, positions, factor_returns,
                                        factor_loadings, transactions)
 
-    # one section for the returns plot, and for each factor grouping
-    # one section for factor returns, and one for risk exposures
-    vertical_sections = 1 + 2 * len(factor_partitions)
+    # only plot partitions for which there are data in perf_attrib_data
+    num_partitions = 0
```

Comment:
This might be made a little clearer by putting a name on the logic for checking valid partitions, e.g.:
```python
def is_valid_partition(partition_name):
    partition_cols = factor_partitions[partition_name]
    intersection = perf_attrib_data.columns.intersection(partition_cols)
    return bool(len(intersection))

num_partitions = sum(map(is_valid_partition, factor_partitions))
```